array.rb

```diff
@@ -1,4 +1,58 @@
 class Array
+  # call-seq:
+  #   array.each {|element| ... } -> self
+  #   array.each -> Enumerator
+  #
+  # Iterates over array elements.
+  #
+  # When a block given, passes each successive array element to the block;
+  # returns +self+:
+  #
+  #   a = [:foo, 'bar', 2]
+  #   a.each {|element|  puts "#{element.class} #{element}" }
+  #
+  # Output:
+  #
+  #   Symbol foo
+  #   String bar
+  #   Integer 2
+  #
+  # Allows the array to be modified during iteration:
+  #
+  #   a = [:foo, 'bar', 2]
+  #   a.each {|element| puts element; a.clear if element.to_s.start_with?('b') }
+  #
+  # Output:
+  #
+  #   foo
+  #   bar
+  #
+  # When no block given, returns a new Enumerator:
+  #   a = [:foo, 'bar', 2]
+  #
+  #   e = a.each
+  #   e # => #<Enumerator: [:foo, "bar", 2]:each>
+  #   a1 = e.each {|element|  puts "#{element.class} #{element}" }
+  #
+  # Output:
+  #
+  #   Symbol foo
+  #   String bar
+  #   Integer 2
+  #
+  # Related: #each_index, #reverse_each.
+  def each
+    unless block_given?
+      return to_enum(:each) { self.length }
+    end
+    i = 0
+    value = nil
+    while Primitive.cexpr!(%q{ ary_fetch_next(self, LOCAL_PTR(i), LOCAL_PTR(value)) })
```

Comment:
https://github.com/ruby/ruby/actions/runs/7524693591/job/20479896887#step:8:21
> <internal:array>:48: warning: assigned but unused variable - i

```suggestion
    _i = 0
    value = nil
    while Primitive.cexpr!(%q{ ary_fetch_next(self, LOCAL_PTR(_i), LOCAL_PTR(value)) })
```